ecl/eclcc/eclcc.cpp

```diff
@@ -1495,9 +1497,15 @@ void EclCC::processFile(EclCompileInstance & instance)
 
     const char * curFilename = instance.inputFile->queryFilename();
     assertex(curFilename);
+    bool inputFromStdIn = streq(curFilename, "stdin:");
+    StringBuffer expandedSourceName;
+    if (!inputFromStdIn && !optNoSourcePath)
+        makeAbsolutePath(curFilename, expandedSourceName);
+    else
+        expandedSourceName.append(curFilename);
 
-    Owned<ISourcePath> sourcePath = optNoSourcePath ? NULL : createSourcePath(curFilename);
-    Owned<IFileContents> queryText = createFileContentsFromFile(curFilename, sourcePath, false, NULL);
+    Owned<ISourcePath> sourcePath = optNoSourcePath ? NULL : createSourcePath(expandedSourceName);
```

Comment:
minor: Is it worth doing (optNoSourcePath || inputFromStdIn) to avoid the extra call ?